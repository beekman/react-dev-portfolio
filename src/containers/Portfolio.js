import React, { Component } from 'react';
import Nav from '../components/Nav/Nav';
import About from '../components/About/About';
import Skills from '../components/Skills/Skills';
import Projects from '../components/Projects/Projects';
import Contact from '../components/Contact/Contact';
import data from '../portfolio-data.js';
export default class Portfolio extends Component {

  render() {
    return (
      <>
        <Nav />
        <About about={data.about} />
        <Skills skills={data.skills} learning={data.learning} />
        <Projects projects={data.projects} />
        <Contact contact={data.contact} />
      </>
    );
  }
}
