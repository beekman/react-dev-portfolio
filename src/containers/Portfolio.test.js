import React from 'react';
import { shallow } from 'enzyme';
import Portfolio from './Portfolio';

describe('Portfolio component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Portfolio />);
    expect(wrapper).toMatchSnapshot();
  });
});
