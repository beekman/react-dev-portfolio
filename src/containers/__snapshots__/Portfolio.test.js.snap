// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches snapshot 1`] = `
<Fragment>
  <Nav />
  <About
    about={
      Object {
        "pitch": "After earning my degree in Multimedia Design at the University of Oregon, I spent more than a decade designing and developing websites. During that time I drilled deeper into full-stack programming and JavaScript.",
        "summary": "I'm a full-stack software engineer with an entrepreneurial background in front-end web development, SEO, and education. I'm enthusiastic about creating code that solves human-centered problems and I love learning and teaching new techniques and technologies.",
        "title": "Creative Software Engineer",
      }
    }
  />
  <Skills
    skills={
      Object {
        "heading": "Tools of My Trade",
        "learning": Array [
          Object {
            "name": "GraphQL",
          },
          Object {
            "name": "D3",
          },
        ],
        "learningCopy": "My tech stack is always growing. Here's what I'm getting to know right now.",
        "resumeLink": "https://www.canva.com/design/DADxzQOaReM/zmkBwZHsYHDKVeaHyVzBoA/view?utm_content=DADxzQOaReM&utm_campaign=designshare&utm_medium=link&utm_source=sharebutton",
        "tech": Array [
          Object {
            "name": "JavaScript",
          },
          Object {
            "name": "TypeScript",
          },
          Object {
            "name": "React",
          },
          Object {
            "name": "Redux",
          },
          Object {
            "name": "Node",
          },
          Object {
            "name": "MongoDB",
          },
          Object {
            "name": "PostgreSQL",
          },
          Object {
            "name": "Express",
          },
          Object {
            "name": "Mongoose",
          },
          Object {
            "name": "Jest",
          },
          Object {
            "name": "TravisCl",
          },
          Object {
            "name": "Git",
          },
          Object {
            "name": "Shell",
          },
          Object {
            "name": "Netlify",
          },
          Object {
            "name": "Heroku",
          },
          Object {
            "name": "MagentaJS",
          },
          Object {
            "name": "ToneJS",
          },
          Object {
            "name": "P5JS",
          },
          Object {
            "name": "ML5",
          },
          Object {
            "name": "Python",
          },
          Object {
            "name": "PHP",
          },
          Object {
            "name": "SASS",
          },
          Object {
            "name": "HTML",
          },
          Object {
            "name": "CSS",
          },
        ],
        "techCopy": "I create digital solutions with a modern tech stack centered around functional, declarative JavaScript. I am always excited to add new tools to that stack.",
        "techHeading": "I use these",
      }
    }
  />
  <Projects
    projects={
      Object {
        "design-portfolio-link": "https://benbeekman.com/web",
        "feature": Array [
          Object {
            "demo": "https://shapeways.netlify.app/",
            "description": "An interactive audiovisual composition, made possible by machine learning, and an exploration of new forms of expression and interaction. I developed Shapeways using React with a team of 5 people for a Machine Learning for Artists hackathon. My personal contributions included functions for real-time interpretation of a user's movement to generate dynamic musical performance seed data, for making API requests to several Magenta.JS musical performance machine learning models using that seed data, and for transforming the API responses into synth output, as well as improvements to the app's UX.",
            "github": "https://github.com/whateverSpace/shapeways-frontend",
            "tech": Array [
              Object {
                "name": "JavaScript",
              },
              Object {
                "name": "React",
              },
              Object {
                "name": "MagentaJS",
              },
              Object {
                "name": "P5JS",
              },
              Object {
                "name": "ML5",
              },
              Object {
                "name": "ToneJS",
              },
              Object {
                "name": "Jest",
              },
              Object {
                "name": "TravisCl",
              },
              Object {
                "name": "GitHub",
              },
              Object {
                "name": "Shell",
              },
              Object {
                "name": "Netlify",
              },
              Object {
                "name": "HTML",
              },
              Object {
                "name": "CSS",
              },
            ],
            "title": "Shapeways",
          },
          Object {
            "demo": "https://live-palette-populator.netlify.app/",
            "description": "A React app I designed and engineered with an uniquely efficient and intuitive UI for quickly creating and exploring flexible color palettes. I engineered it with experimentation and efficiency in mind, as a tool for quickly generating and exporting large or small collections of colors using any combination of color relationships and color modes.",
            "github": "https://github.com/beekman/live-palette-populator",
            "tech": Array [
              Object {
                "name": "JavaScript",
              },
              Object {
                "name": "React",
              },
              Object {
                "name": "Jest",
              },
              Object {
                "name": "TravisCl",
              },
              Object {
                "name": "GitHub",
              },
              Object {
                "name": "Shell",
              },
              Object {
                "name": "Netlify",
              },
              Object {
                "name": "HTML",
              },
              Object {
                "name": "CSS",
              },
            ],
            "title": "Live Palette Populator",
          },
          Object {
            "demo": "https://whateversynthstaging.netlify.app/",
            "description": "A JavaScript-based synthesizer app featuring MIDI and keyboard support, an unlimited effects chain with over 10 audio effects using tuna.js, pitch bend and modulation, instrument selection, and real-time waveform visualization. I built this app with three fellow musicians. Some of my most significant personal contributions relate to the user interface and UX engineering. I wrote and implemented many of the UI-oriented React components and hooks, including UI components used throughout the app that make manipulation of real-time effects more efficient, and most of the audio effects components themselves.",
            "github": "https://github.com/whateverSynth/whateverSynth-frontend",
            "tech": Array [
              Object {
                "name": "JavaScript",
              },
              Object {
                "name": "React",
              },
              Object {
                "name": "Jest",
              },
              Object {
                "name": "GitHub",
              },
              Object {
                "name": "TravisCl",
              },
              Object {
                "name": "Shell",
              },
              Object {
                "name": "Netlify",
              },
              Object {
                "name": "HTML",
              },
              Object {
                "name": "CSS",
              },
            ],
            "title": "whateverSynth",
          },
          Object {
            "demo": "https://friendlier-staging.netlify.app",
            "description": "A progressive web app for improving communication, with customizable scheduled push notifications and connection logs. Developed in a five person team over a four-day code sprint. My contributions centered around UX/UI engineering, the creation of React components for adding and editing contacts and a responsive navigation menu system.",
            "github": "https://github.com/band-m/friendlier-fe",
            "tech": Array [
              Object {
                "name": "JavaScript",
              },
              Object {
                "name": "React",
              },
              Object {
                "name": "Redux",
              },
              Object {
                "name": "Node",
              },
              Object {
                "name": "Express",
              },
              Object {
                "name": "MongoDB",
              },
              Object {
                "name": "Jest",
              },
              Object {
                "name": "GitHub",
              },
              Object {
                "name": "Shell",
              },
              Object {
                "name": "Netlify",
              },
              Object {
                "name": "Heroku",
              },
              Object {
                "name": "HTML",
              },
              Object {
                "name": "CSS",
              },
            ],
            "title": "kit: Keep In Touch",
          },
          Object {
            "demo": "https://plantonomous.herokuapp.com",
            "description": "A secure server-side IoT plant sensor application that allows users to remotely gather and post data collected from a variety of sensors (light, temperature, humidity) via a Raspberry Pi. Two-factor authentication ensures the integrity of data. Data can be interpreted to make recommendations of common house plants that may thrive under matching environmental conditions.",
            "github": "https://github.com/piParty/pi-party",
            "tech": Array [
              Object {
                "name": "JavaScript",
              },
              Object {
                "name": "Node",
              },
              Object {
                "name": "MongoDB",
              },
              Object {
                "name": "Express",
              },
              Object {
                "name": "Mongoose",
              },
              Object {
                "name": "TravisCl",
              },
              Object {
                "name": "GitHub",
              },
              Object {
                "name": "Shell",
              },
              Object {
                "name": "HTML",
              },
              Object {
                "name": "CSS",
              },
              Object {
                "name": "Python",
              },
              Object {
                "name": "Flask",
              },
              Object {
                "name": "Raspberry Pi",
              },
              Object {
                "name": "Heroku",
              },
            ],
            "title": "Plant Parenthood",
          },
          Object {
            "demo": "https://front-pagenews.herokuapp.com",
            "description": "A Tinder-style newsreader that allows you to swipe right on the news that interests you, and filter out news sources that you don't want to see.",
            "github": "https://github.com/Alchemy-Frontpage/Frontpage",
            "tech": Array [
              Object {
                "name": "JavaScript",
              },
              Object {
                "name": "Node",
              },
              Object {
                "name": "Express",
              },
              Object {
                "name": "PostgreSQL",
              },
              Object {
                "name": "GitHub",
              },
              Object {
                "name": "TravisCl",
              },
              Object {
                "name": "Shell",
              },
              Object {
                "name": "HTML",
              },
              Object {
                "name": "CSS",
              },
            ],
            "title": "FrontPage",
          },
          Object {
            "demo": "https://i-16.github.io/qwixx/",
            "description": "A mobile-responsive vanilla JavaScript implementation of the dice game Qwixx that speeds gameplay and doesn't waste scorecards. Developed with face-to-face multiplayer in mind, with each player's device acting as their scorecard.",
            "github": "https://github.com/I-16/qwixx",
            "tech": Array [
              Object {
                "name": "JavaScript",
              },
              Object {
                "name": "TravisCl",
              },
              Object {
                "name": "HTML",
              },
              Object {
                "name": "CSS",
              },
              Object {
                "name": "GitHub",
              },
              Object {
                "name": "Shell",
              },
            ],
            "title": "Qwixx",
          },
        ],
        "history": "I'm fascinated by the ways digital technology can engage and empower people. Since earning my U of O degree in Multimedia Design, I've spent more than a decade helping people harness the power of that technology. I've taught classes for young people, tutored older people, consulted for small businesses, co-authored technology textbooks, served as both design and IT department for a small media company, and designed and developed web sites for businesses, nonprofits, and individuals.",
        "summary": "I work well on my own, but I thrive in collaborative coding environments. Here's a sampling of both my individual and team-based projects.",
      }
    }
  />
  <Contact
    contact={
      Object {
        "city": "Portland",
        "email": "ben@benbeekman.dev",
        "githubURL": "https://github.com/beekman",
        "heading": "Let's Connect!",
        "hiringStatus": "You can reach me using any of the contact options below.",
        "linkedinURL": "https://www.linkedin.com/in/beekman/",
        "name": "Ben Beekman",
        "occupation": "Software Engineer",
        "phone": "(503)708-0478",
        "resumeLink": "https://www.canva.com/design/DAEMyW-q7Kw/2QtY4oerkRr8TTz2owsmyQ/view?utm_content=DAEMyW-q7Kw&utm_campaign=designshare&utm_medium=link&utm_source=publishsharelink",
        "state": "OR",
        "tagline": "import SoftwareEngineer from 'benbeekman.dev';",
      }
    }
  />
</Fragment>
`;
